scripts/lib.sh

```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env bash
+# Common functions for Kubeflow manifest synchronization scripts
+
+setup_error_handling() {
+  set -euxo pipefail
+  IFS=$'\n\t'
+}
+
+# Check if the git repository has uncommitted changes
+check_uncommitted_changes() {
+  if [ -n "$(git status --porcelain)" ]; then
+    echo "WARNING: You have uncommitted changes"
+  fi
+}
+
+# Create a new git branch if it doesn't exist
+create_branch() {
+  local branch="$1"
+  
+  check_uncommitted_changes
+  
+  if [ $(git branch --list "$branch") ]; then
+    echo "WARNING: Branch $branch already exists."
+  fi
+  
+  if ! git show-ref --verify --quiet refs/heads/$branch; then
+    git checkout -b "$branch"
+  else
+    echo "Branch $branch already exists."
+  fi
+}
+
+clone_and_checkout() {
+  local src_dir="$1"
+  local repo_url="$2"
+  local repo_dir="$3"
+  local commit="$4"
+  
+  echo "Checking out in $src_dir to $commit..."
+  
+  mkdir -p "$src_dir"
+  cd "$src_dir"
+  
+  # Clone repository if it doesn't exist
+  if [ ! -d "$repo_dir/.git" ]; then
+    git clone "$repo_url" "$repo_dir"
+  fi
+  
+  # Checkout to specific commit
+  cd "$src_dir/$repo_dir"
+  if ! git rev-parse --verify --quiet "$commit"; then
+    git checkout -b "$commit"
+  else
+    git checkout "$commit"
+  fi
+  
+  check_uncommitted_changes
+}
+
+# Copy manifests from source to destination
+copy_manifests() {
+  local src="$1"
```

Comment:
please use full variable names source and destination. It should be english text in the end.